src/org/pentaho/reporting/platform/plugin/JobManager.java

```diff
@@ -111,15 +114,46 @@ public JobManager( boolean isSuportAsync ) {
     StreamingOutput stream = new StreamingOutputWrapper( input );
 
     MediaType mediaType = null;
+    Response.ResponseBuilder response = null;
     try {
       mediaType = MediaType.valueOf( state.getMimeType() );
+      response = Response.ok( stream, mediaType );
     } catch ( Exception e ) {
       logger.error( "can't determine JAX-RS media type for: " + state.getMimeType() );
-      // may be this will work?
-      return Response.ok( stream, state.getMimeType() ).build();
+      response = Response.ok( stream, state.getMimeType() );
     }
 
-    return Response.ok( stream, mediaType ).build();
+    // no cache
+    CacheControl cacheControl = new CacheControl();
+    cacheControl.setPrivate( true );
+    cacheControl.setMaxAge( 0 );
+    cacheControl.setMustRevalidate( true );
+
+    response.cacheControl( cacheControl );
+
+    response = calculateContentDisposition( response, state );
+
+    return response.build();
+  }
+
+  private Response.ResponseBuilder calculateContentDisposition( Response.ResponseBuilder response,
+      IAsyncReportState state ) {
+    String extension = MimeHelper.getExtension( state.getMimeType() );
+    String path = state.getPath();
+    String fileName = path.substring( path.lastIndexOf( "/" ) + 1, path.lastIndexOf( "." ) );
```

Comment:
Please make this code safe. Your path may nor have a "/" or "." character. If you want a safe option for that, use IOUtils#getFileName and IOUtils#stripFileExtension from libbase. It contains the necessary checks to handle the -1 index returned by "lastIndexOf" in case the search string is not found.